### ISSUE TITLE
[Bug] Assertion `tapePos == playImage->getEndTime()' failed. [sf#540]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#931]
**Reported by manuelbi on 2014-06-20 20:34 UTC**
I was trying to load the CD Sequential stuff again. This requires to turn of motorcontrol of the cassetteplayer. I turned it off, let it run to the end and then changed cassette image. Then I got this assert:
openmsx: src/cassette/CassettePlayer.cc:618: virtual void openmsx::CassettePlayer::executeUntil(param, int): Assertion `tapePos == playImage->getEndTime()' failed.
Aborted (core dumped)

Playing around with cchanging cassettes with motor control off I got it reproduced.

stack info:

```
#0  0x00007fbade420407 in __GI_raise (sig=sig@entry=6)
    at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007fbade423508 in __GI_abort () at abort.c:89
#2  0x00007fbade419516 in __assert_fail_base (
    fmt=0x7fbade54fc60 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x7a3a58 "tapePos == playImage->getEndTime()", 
    file=file@entry=0x7a3988 "src/cassette/CassettePlayer.cc", 
    line=line@entry=618, 
    function=function@entry=0x7a4580 <openmsx::CassettePlayer::executeUntil(openmsx::EmuTime, int)::__PRETTY_FUNCTION__> "virtual void openmsx::CassettePlayer::executeUntil(param, int)") at assert.c:92
#3  0x00007fbade4195c2 in __GI___assert_fail (
    assertion=0x7a3a58 "tapePos == playImage->getEndTime()", 
    file=0x7a3988 "src/cassette/CassettePlayer.cc", line=618, 
    function=0x7a4580 <openmsx::CassettePlayer::executeUntil(openmsx::EmuTime, int)::__PRETTY_FUNCTION__> "virtual void openmsx::CassettePlayer::executeUntil(param, int)") at assert.c:101
#4  0x00000000004676c8 in openmsx::CassettePlayer::executeUntil (
    this=0x1abfae0, time=..., userData=<optimized out>)
    at src/cassette/CassettePlayer.cc:618
#5  0x000000000045a08e in openmsx::Scheduler::scheduleHelper (
    this=this@entry=0xea5a90, limit=..., limit@entry=...)
    at src/Scheduler.cc:158
#6  0x00000000004d0c49 in schedule (limit=..., this=0xea5a90)
    at src/Scheduler.hh:68
#7  openmsx::CPUCore<openmsx::Z80TYPE>::execute2 (this=this@entry=0x1aa4c30, 
    fastForward=<optimized out>) at src/cpu/CPUCore.cc:2561
#8  0x00000000004d0cee in openmsx::CPUCore<openmsx::Z80TYPE>::execute (
    this=0x1aa4c30, fastForward=<optimized out>) at src/cpu/CPUCore.cc:2525
#9  0x000000000050db5f in openmsx::MSXCPU::execute (this=<optimized out>, 
    fastForward=fastForward@entry=false) at src/cpu/MSXCPU.cc:163
#10 0x000000000042f954 in openmsx::MSXMotherBoard::Impl::execute (
    this=<optimized out>) at src/MSXMotherBoard.cc:716
#11 0x00000000004490a1 in openmsx::Reactor::run (
    this=this@entry=0x7fff1baa4bb0, parser=...) at src/Reactor.cc:612
#12 0x000000000041096c in openmsx::main (argc=<optimized out>, 
    argv=<optimized out>) at src/main.cc:153
#13 0x0000000000410c59 in main (argc=<optimized out>, argv=<optimized out>)
    at src/main.cc:182
```

**Comment by openMSX-import** [comment]
**Updated by m9710797 on 2014-06-24 19:28 UTC**
- **status**: open --> closed-fixed

**Comment by openMSX-import** [comment]
**Commented by m9710797 on 2014-06-24 19:28 UTC**
Fixed in 9dba46cf50.
